File: test_data/code/java/1592-rearrange-spaces-between-words.java
class Solution {
    public String reorderSpaces(String text) {
        String[] words = text.trim().split("\\s+");
        int totalSpaces = text.length() - text.trim().length();
        int wordsCount = words.length;
        
        if (wordsCount == 1) {
            return words[0] + String.format("%" + totalSpaces + "s", " ").replace(" ", "");
        }
        
        int spacesBetweenWords = totalSpaces / (wordsCount - 1);
        int leftOverSpaces = totalSpaces % (wordsCount - 1);
        
        StringBuilder result = new StringBuilder();
        for (String word : words) {
            result.append(word);
            if (result.length() < text.length()) {
                for (int i = 0; i < spacesBetweenWords; i++)
                    result.append(' ');
                if (leftOverSpaces > 0) {
                    result.append(' ');
                    leftOverSpaces--;
                }
            }
        }
        return result.toString();
    }
}

File: test_data/code/python/1592-rearrange-spaces-between-words.py
Python
class Solution:
    def reorderSpaces(self, text: str) -> str:
        words = text.split()
        num_of_words = len(words)
        total_spaces = text.count(' ')
        
        if num_of_words == 1:
            extra_spaces = total_spaces
            space_between_words = 0
        else:
            extra_spaces = total_spaces - (num_of_words - 1)
            space_between_words = total_spaces // (num_of_words - 1)
        
        result = ' '.join(words)
        if extra_spaces > 0:
            result += ' ' * extra_spaces
        
        return result